code/modules/mob/death.dm

```diff
@@ -28,13 +28,12 @@
 //This is the proc for turning a mob into ash. Mostly a copy of gib code (above).
 //Originally created for wizard disintegrate. I've removed the virus code since it's irrelevant here.
 //Dusting robots does not eject the MMI, so it's a bit more powerful than gib() /N
-/mob/proc/dust()
-	death(1)
+/mob/proc/dust(visual_only = FALSE)
+	if(!visual_only)
+		death(1)
+		icon = null
+		invisibility = 101
```

Comment:
Would it be a better idea to split the dust animation into its own proc, and call it here, instead?